src/SchemaDesign/SchemaDesigner.tsx

```diff
@@ -0,0 +1,416 @@
+                <></>
+              </SchemaForm>
+            )}
+          </Paper>
+        </Grid>
+      </Grid>
+      <Divider orientation='horizontal' flexItem sx={{ mt: 4, mb: 4 }} />
+      <Box sx={{ textAlign: 'right' }}>
+        <Button variant='contained' onClick={() => setSubmitSchemaDialogOpen(true)} disabled={steps.length === 0}>
+          Submit schema
+        </Button>
+      </Box>
+      <QuestionPicker
+        onSubmit={addQuestion}
+        handleClose={handleQuestionPickerClose}
+        questionPickerOpen={questionPickerOpen}
+      />
+      <Dialog open={newStepDialogOpen} onClose={() => setNewStepDialogOpen(false)}>
+        <DialogTitle>Add new Step</DialogTitle>
+        <Box component='form' onSubmit={addNewStep}>
+          <DialogContent>
+            <DialogContentText>
+              The step name and reference must be unique, if it is left blank it will be generated automatically using
+              the step name. The step name is the title displayed on the form itself.
+            </DialogContentText>
+            <Stack direction='row' spacing={2} sx={{ pt: 2 }}>
+              <TextField
+                label='Step reference'
+                onChange={(event): void => setStepReference(event.target.value)}
+                value={stepReference}
+              />
+              <Stack>
+                <TextField
+                  required
+                  label='Step name'
+                  onChange={(event): void => setStepName(event.target.value)}
+                  value={stepName}
+                />
+                <Typography variant='caption' sx={{ color: '#e35a5a' }}>
+                  {stepErrorText}
+                </Typography>
+              </Stack>
+            </Stack>
+          </DialogContent>
+          <DialogActions>
+            <Button onClick={() => setNewStepDialogOpen(false)}>Cancel</Button>
+            <Button type='submit'>Add Step</Button>
+          </DialogActions>
+        </Box>
+      </Dialog>
+      <Dialog open={submitSchemaDialogOpen} onClose={() => setSubmitSchemaDialogOpen(false)}>
+        <DialogTitle>Submit schema</DialogTitle>
+        <DialogContent>
+          <DialogContentText>
+            The schema name and reference must be unique, if the reference is left blank it will be generated
+            automatically using the schema name.
+          </DialogContentText>
+          <Stack spacing={1}>
+            <TextField
+              label='Schema reference'
+              onChange={(event): void => setSchemaReference(event.target.value)}
+              value={schemaReference}
+            />
+            <TextField
+              required
+              label='Schema name'
+              onChange={(event): void => setSchemaName(event.target.value)}
+              value={schemaName}
+            />
+            <InputLabel id='schema-use-label'>Schema Type</InputLabel>
+            <Select
+              labelId='schema-use-label'
```

Comment:
I don't think these `labelId` and `id` props are needed

Reply:
I'm under the impression it's needed to tie the label and select component together for accessibility reasons, but happy to remove if this is not the case

Reply:
We do need the `labelid` on the `Select` but we don't need the `id='demo-simple-select'` below